data-products/src/main/scala/org/sunbird/analytics/job/report/CollectionSummaryJob.scala

```diff
@@ -0,0 +1,183 @@
+import org.ekstep.analytics.framework.util.DatasetUtil.extensions
+import org.ekstep.analytics.framework.util.{CommonUtil, JSONUtils, JobLogger}
+import org.ekstep.analytics.framework.{FrameworkContext, IJob, JobConfig}
+import org.joda.time.format.{DateTimeFormat, DateTimeFormatter}
+import org.joda.time.{DateTime, DateTimeZone}
+import org.sunbird.analytics.util.{CourseUtils, DecryptUtil, UserData}
+
+import scala.collection.immutable.List
+
+
+case class CollectionBatch(batchId: String, courseId: String, startDate: String, endDate: String)
+
+case class CourseMetrics(processedBatches: Option[Int], failedBatches: Option[Int], successBatches: Option[Int])
+
+case class CollectionBatchResponses(batchId: String, execTime: Long, status: String)
+
+object CollectionSummaryJob extends optional.Application with IJob with BaseReportsJob {
+  val cassandraUrl = "org.apache.spark.sql.cassandra"
+  private val userCacheDBSettings = Map("table" -> "user", "infer.schema" -> "true", "key.column" -> "userid")
+  private val userEnrolmentDBSettings = Map("table" -> "user_enrolments", "keyspace" -> AppConf.getConfig("sunbird.courses.keyspace"), "cluster" -> "LMSCluster")
+  private val courseBatchDBSettings = Map("table" -> "course_batch", "keyspace" -> AppConf.getConfig("sunbird.courses.keyspace"), "cluster" -> "LMSCluster")
+
+  implicit val className: String = "org.sunbird.analytics.job.report.CollectionSummaryJob"
+  val jobName = "CollectionSummaryJob"
+
+  private val columnsOrder = List(
+    "Published by", "Batch id", "Collection id",
+    "Collection name", "Batch start date", "Batch end date",
+    "State", "Total enrolments By State", "Total completion By State",
+    "Has certificate", "Total Certificate issued by State");
+
+  private val columnMapping = Map("batchid" -> "Batch id",
+    "publishedBy" -> "Published by",
+    "courseid" -> "Collection id", "collectionName" -> "Collection name",
+    "startdate" -> "Batch start date",
+    "enddate" -> "Batch end date",
+    "hasCertified" -> "Has certificate",
+    "enrolledUsersCountByState" -> "Total enrolments By State",
+    "completionUserCountByState" -> "Total completion By State",
+    "userstate" -> "State",
+    "certificateIssueCount" -> "Total Certificate issued by State"
+  )
+
+  // $COVERAGE-OFF$ Disabling scoverage for main and execute method
+  override def main(config: String)(implicit sc: Option[SparkContext] = None, fc: Option[FrameworkContext] = None) {
+    JobLogger.init(jobName)
+    JobLogger.start(s"$jobName started executing", Option(Map("config" -> config, "model" -> jobName)))
+    implicit val jobConfig: JobConfig = JSONUtils.deserialize[JobConfig](config)
+    implicit val spark: SparkSession = openSparkSession(jobConfig)
+    implicit val frameworkContext: FrameworkContext = getReportingFrameworkContext()
+    init()
+    val conf = config.split(";")
+    val batchIds = if (conf.length > 1) {
+      conf(1).split(",").toList
```

Comment:
config should be similar to exhaust jobs. We should have capability to run for batches matching a search query